contracts/Governance/VestingToken.sol

```diff
@@ -0,0 +1,115 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.7.5;
+
+/// Token vesting contract.
+/// Adapted from Melonport AG <team@melonport.com>
+
+import "@openzeppelin/contracts/token/ERC20/ERC20.sol";
+
+contract VestingToken {
+    using SafeMath for uint256;
+
+    ERC20 public immutable token; // Radicle ERC20 contract
+
+    address public immutable owner; // deployer; can interrupt vesting
+    uint256 public immutable totalVestingAmount; // quantity of vested token in total
+    uint256 public immutable vestingStartTime; // timestamp when vesting is set
+    uint256 public immutable vestingPeriod; // total vesting period in seconds
+    address public immutable beneficiary; // address of the beneficiary
+
+    bool public interrupted; // whether vesting is still possible
+    uint256 public withdrawn; // quantity of token withdrawn so far
+
+    modifier notInterrupted() {
+        require(!interrupted, "The contract has been interrupted");
+        _;
+    }
+
+    modifier onlyOwner() {
+        require(msg.sender == owner, "Only owner can do this");
+        _;
+    }
+
+    modifier onlyBeneficiary() {
+        require(msg.sender == beneficiary, "Only beneficiary can do this");
+        _;
+    }
+
+    /// @notice Create a vesting allocation of tokens.
+    ///
+    /// @param _token Address of token being vested
+    /// @param _beneficiary Address of beneficiary
+    /// @param _amount Amount of tokens
+    /// @param _vestingPeriod Vesting period in seconds from vestingStartTime
+    /// @param _vestingStartTime Vesting start time in seconds since Epoch
+    constructor(
+        address _token,
+        address _owner,
```

Comment:
With the current API you can create the vesting contract without becoming an owner. Is it intended? It raises a flag when reading, because then you'll give the vested tokens, but the owner will take them if terminated.

Reply:
I think that's fine. This allows for not having an owner too, which is useful, eg. when granting vesting tokens to the community.

Reply:
Cool trick!